src/__tests__/rules.test.ts

```diff
@@ -76,4 +82,54 @@ describe('rules', () => {
       expect(() => require(`../rules/${ruleName}`)).not.toThrow();
     });
   });
+
+  describe('rule documentation files have the correct content', () => {
+    it.each(ruleNames)('%s', ruleName => {
```

Comment:
I don't really think we need this test because we're able to re-run the docs tool in ci and check if any files have changed, whereas we'd have to update this test if the output changes and its relatively complex.

The only thing we won't catch from the tool is if we've got or not got a header for options, but I'm not super fussed about that because that's a pretty obvious one and the more important thing is ensuring all the options are documented which is a lot harder to automate. (it'd also be pretty easy to have the docs tool check & fail on anyway).

Reply:
I removed the tests for the content that is automatically generated.

I do want to continue to advocate for testing that:
* incorrect notices are not present
* section title consistency like "Rule details"
* "Options" section is included only for rules with options

Not only did these tests catch many issues with the jest rule docs, but they also save a lot of time during PR reviews, since many contributors will often just copy and paste an existing rule doc, and then PR reviewers will have to remind them to remove all the outdated content (and outdated content/notices can easily slip through). These tests have saved me a lot of time in my other plugins.

I also have some tests in other plugins that ensure that each configuration option is mentioned in the rule doc like [this example](https://github.com/ember-cli/eslint-plugin-ember/blob/cb145d8f1b37cfa940f27ce57b0ffe45dc85ab4b/tests/rule-setup.js#L133). That would be a good follow-up to this PR, so I do think we can and should have at least basic tests that rule docs document their rule options.

I would also argue that having a few actual jest tests for rule docs is preferable to having the rule doc generator fail. Jest tests/assertions are standardized, easy/convenient to write, and will provide more detailed and consistent failure output than some custom failure in the rule doc generator.

Reply:
For now I disagree: most of what you've put is already covered [by us running the docs tool in CI](https://github.com/jest-community/eslint-plugin-jest/blob/main/.github/workflows/nodejs.yml#L103-L121), with the resulting output telling people what they need to do to resolve the problem (run the tool).

My dislike is with how brittle this still feels + you needing to export stuff from `tools/`, vs our current CI setup.

Reply:
I agree `tools/` isn't the best place to export stuff from so I can definitely move `rule-notices.ts` to `lib/` instead if you prefer.

But I'm confused by "most of what you've put is already covered [by us running the docs tool in CI](https://github.com/jest-community/eslint-plugin-jest/blob/main/.github/workflows/nodejs.yml#L103-L121)" as I've explicitly removed the tests for auto-generated content and only left in place tests that basic details are included or not included in the rule doc (which cannot be resolved by running the tool).

Reply:
I would expect the tool to be able to handle the notices completely, since if we can detect them we should be able to remove them as well.

As the other stuff it feels very low gain because the important part is actually _where_ in the document they are - i.e. you can make the tests pass by just adding those headers at the start or end. Personally what I'd rather see from this is actually further evolving the docs tool (e.g. have it output warnings about what you're testing for here) because that's more portable - we actually should already be almost at a point where you could have it as its own package (we'd just need to sort out loading the rules).

But that's out of scope for this I think - the key thing for me here is that we're still checking about the notices when we should be able to handle that in the tool.

Reply:
Okay, I went ahead and moved these tests (checking for `## Rule details` and `## Options`) into the generator. If these sections are incorrectly present, we will output the issue and the script will exit with failure.